src/osd/ECBackend.cc

```diff
@@ -1541,6 +1541,42 @@ void ECBackend::dump_recovery_info(Formatter *f) const
   f->close_section();
 }
 
+struct ECClassicalOp : ECBackend::Op {
+  PGTransactionUPtr t;
+
+  void generate_transactions(
+      ECTransaction::WritePlan &plan,
+      ceph::ErasureCodeInterfaceRef &ecimpl,
+      pg_t pgid,
+      const ECUtil::stripe_info_t &sinfo,
+      const std::map<hobject_t,extent_map> &partial_extents,
+      std::vector<pg_log_entry_t> &entries,
+      std::map<hobject_t,extent_map> *written,
+      std::map<shard_id_t, ObjectStore::Transaction> *transactions,
+      std::set<hobject_t> *temp_added,
+      std::set<hobject_t> *temp_removed,
+      DoutPrefixProvider *dpp,
+      const ceph_release_t require_osd_release) override
+  {
+    if (t) {
```

Comment:
Can t actually be null?

Reply:
Nope, it's already dereferencec without extra checks.